apps/frontend/src/pages/auth/libs/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,15 +1,62 @@
-import { Button } from "~/libs/components/components.js";
+import { Button, Input, Link } from "~/libs/components/components.js";
+import { AppRoute } from "~/libs/enums/enums.js";
+import { useAppForm, useCallback } from "~/libs/hooks/hooks.js";
+import {
+	type UserSignInRequestDto,
+	userSignInValidationSchema,
+} from "~/modules/users/users.js";
+
+import { DEFAULT_SIGN_IN_PAYLOAD } from "./libs/constants/constants.js";
+import styles from "./styles.module.css";
 
 type Properties = {
-	onSubmit: () => void;
+	onSubmit: (payload: UserSignInRequestDto) => void;
 };
 
 const SignInForm = ({ onSubmit }: Properties): React.JSX.Element => {
+	const { control, errors, handleSubmit } = useAppForm<UserSignInRequestDto>({
+		defaultValues: DEFAULT_SIGN_IN_PAYLOAD,
+		validationSchema: userSignInValidationSchema,
+	});
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			void handleSubmit((formData: UserSignInRequestDto) => {
+				onSubmit(formData);
+			})(event_);
+		},
+		[handleSubmit, onSubmit],
+	);
+
 	return (
-		<form noValidate onSubmit={onSubmit}>
-			<h1>Sign In</h1>
-			<Button label="Sign in" />
-		</form>
+		<div className={styles["container"]}>
+			<h1 className={styles["form-title"]}>Log In</h1>
+			<form className={styles["form"]} onSubmit={handleFormSubmit}>
```

Comment:
It should work like this 🤔 
```suggestion
			<form className={styles["form"]} onSubmit={handleSubmit(onSubmit)}>
```

Reply:
> It should work like this 🤔

```
Promise-returning function provided to attribute where a void return was expected  @typescript-eslint/no-misused-promises
```
i think that in the sign-up form, this was also done precisely for this reason

Reply:
> It should work like this 🤔
``` tsx
import { Button, Input, Link } from "~/libs/components/components.js";
import { AppRoute } from "~/libs/enums/enums.js";
import { useAppForm } from "~/libs/hooks/hooks.js";
import {
	type UserSignInRequestDto,
	userSignInValidationSchema,
} from "~/modules/users/users.js";

import { DEFAULT_SIGN_IN_PAYLOAD } from "./libs/constants/constants.js";
import styles from "./styles.module.css";

type Properties = {
	onSubmit: (payload: UserSignInRequestDto) => void;
};

const SignInForm = ({ onSubmit }: Properties): React.JSX.Element => {
	const { control, errors, handleSubmit } = useAppForm<UserSignInRequestDto>({
		defaultValues: DEFAULT_SIGN_IN_PAYLOAD,
		validationSchema: userSignInValidationSchema,
	});

	return (
		<div className={styles["container"]}>
			<h1 className={styles["form-title"]}>Log In</h1>
			<form className={styles["form"]} onSubmit={handleSubmit(onSubmit)}>
```

```
 error  Promise-returning function provided to attribute where a void return was expected  @typescript-eslint/no-misused-promises
 ```
 It seems that what we have now is the only correct option

Reply:
I'll wait, thx

Reply:
Yeah, only to update this place, and how redirect works on Auth component

And when sign up PR will be merged, to resolve conflicts. Other than that, all looks clean in my opinion ✨